controllers/object_controls.go

```diff
@@ -1327,6 +1337,11 @@ func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec,
 		setContainerEnv(mainContainer, "CONTAINERD_RUNTIME_CLASS", getRuntimeClass(config))
 	}
 
+	if runtime == gpuv1.CRIO.String() {
+		// We add the nvidia runtime to the cri-o config by default instead of installing the OCI prestart hook
+		setContainerEnv(mainContainer, CrioConfigModeEnvName, "config")
```

Comment:
Question: Why are we setting this twice?

Reply:
You are right, this envvar is getting set here and in `transformForCDI()`. I believe I can remove it from `transformForCDI()` now that we are always defaulting to the `config` mode for cri-o here...

Reply:
However the unit tests confirm that the resulting pod spec only has this envvar set once. It is because `setContainerEnv()` handles duplicates -- it will override the value for an existing key instead of blindly appending envvars to the list.